PIP_COMPATIBILITY.md

```diff
@@ -0,0 +1,317 @@
+the implementation, and tend to be tracked in individual issues. For example:
+
+- [`--trusted-host`](https://github.com/astral-sh/uv/issues/1339)
+- [`--user`](https://github.com/astral-sh/uv/issues/2077)
+- [`--no-build-isolation`](https://github.com/astral-sh/uv/issues/1715)
+
+If you encounter a missing option or subcommand, please search the issue tracker to see if it has
+already been reported, and if not, consider opening a new issue. Feel free to upvote any existing
+issues to convey your interest.
+
+## Resolution strategy
+
+For a given set of dependency specifiers, it's often the case that there is no single "correct"
+set of packages to install. Instead, there are many valid sets of packages that satisfy the
+specifiers.
+
+Neither `pip` nor uv make any guarantees about the _exact_ set of packages that will be
+installed; only that the resolution will be consistent, deterministic, and compliant with the
+specifiers. As such, in some cases, `pip` and uv will yield different resolutions; however, both
+resolutions _should_ be equally valid.
+
+For example, consider:
+
+```txt
+# requirements.txt
+starlette
+fastapi
+```
+
+At time of writing, the most recent `starlette` version is `0.37.2`, and the most recent `fastapi`
+version is `0.110.0`. However, `fastapi==0.110.0` also depends on `starlette`, and introduces an
+upper bound: `starlette>=0.36.3,<0.37.0`.
+
+If a resolver prioritizes including the most recent version of `starlette`, it would need to use
+an older version of `fastapi` that excludes the upper bound on `starlette`. In practice, this
+requires falling back to `fastapi==0.1.17`:
+
+```txt
+# This file was autogenerated by uv via the following command:
+#    uv pip compile -
+annotated-types==0.6.0
+    # via pydantic
+anyio==4.3.0
+    # via starlette
+fastapi==0.1.17
+idna==3.6
+    # via anyio
+pydantic==2.6.3
+    # via fastapi
+pydantic-core==2.16.3
+    # via pydantic
+sniffio==1.3.1
+    # via anyio
+starlette==0.37.2
+    # via fastapi
+typing-extensions==4.10.0
+    # via
+    #   pydantic
+    #   pydantic-core
+```
+
+Alternatively, if a resolver prioritizes including the most recent version of `fastapi`, it would
+need to use an older version of `starlette` that satisfies the upper bound. In practice, this
+requires falling back to `starlette==0.36.3`:
+
+```txt
+#    uv pip compile -
+annotated-types==0.6.0
+    # via pydantic
+anyio==4.3.0
+    # via starlette
+fastapi==0.110.0
+idna==3.6
+    # via anyio
+pydantic==2.6.3
+    # via fastapi
+pydantic-core==2.16.3
+    # via pydantic
+sniffio==1.3.1
+    # via anyio
+starlette==0.36.3
+    # via fastapi
+typing-extensions==4.10.0
+    # via
+    #   fastapi
+    #   pydantic
+    #   pydantic-core
+```
+
+When uv resolutions differ from `pip` in undesirable ways, it's often a sign that the specifiers
+are too loose, and that the user should consider tightening them. For example, in the case of
```

Comment:
I'd make this more concrete: Please add `>=` bounds to your packages.

Reply:
Isn't this clear in the example?